pkg/channel/distributed/dispatcher/dispatcher/handler.go

```diff
@@ -130,8 +131,9 @@ func (h *Handler) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama
 // Consume A Single Message
 func (h *Handler) consumeMessage(context context.Context, consumerMessage *sarama.ConsumerMessage, destinationURL *url.URL, replyURL *url.URL, deadLetterURL *url.URL, retryConfig *kncloudevents.RetryConfig) error {
 
+	debug := h.Logger.Core().Enabled(zap.DebugLevel)
 	// Debug Log Kafka ConsumerMessage
-	if h.Logger.Core().Enabled(zap.DebugLevel) {
+	if debug {
```

Comment:
do we really need this check ? and instead do direct `logger.debug` ? 
e.g. https://github.com/knative-sandbox/eventing-kafka/blob/main/pkg/channel/consolidated/reconciler/controller/controller.go#L110

Reply:
It's not usually necessary, but the goal for this particular use of the debug check is to prevent having to call functions like StringifyHeaderPtrs and formatDispatcherExecutionInfo when debug logging isn't enabled.  The zap.() functions create structs that are clever enough to not call the reflection functions if the log level isn't set, but, e.g. StringifyHeaderPtrs isn't; it would be called even when debug is not enabled, before the Logger.Debug call is made, which is a waste of time in consumeMessage, which is data-plane and execution time is (typically) worth the consideration.  I am open to doing this in a different way (although this particular check was here before this PR, I just moved it to a temp variable so I could use it before calling formatDispatcherExecutionInfo) but I don't want to call these debugging functions in ordinary (production) cases.

Reply:
(I'd have used the zap.* functions instead but there are no zap structs for "convert these base64-encoded headers into unencoded text" or "truncate the body of the response if over 500 chars"  ...  although I haven't looked into whether it's possible to add custom zap fields - that might be interesting)

Reply:
It looks like it is possible to add a zapcore.ObjectMarshaler for a type of your own (or a wrapper you make for a third-party object) but I don't know that doing that would add a lot of value here for one function.

Reply:
thanks for the context

/lgtm

leaving the actual approval for @travis-minke-sap